### DISCUSSION TITLE
How is phantom-types different from beartype?

### DISCUSSION THREAD

**Discussion by majidaldo** [antonagestam/phantom-types#211] [Q&A] [answered]
how is this functionally different from [beartype](https://github.com/beartype)? in beartype you can check `Annotated[basetype, <predicates>]`.

**Comment by antonagestam** [comment, marked as answer]
@majidaldo Hi 👋

Generally speaking beartype is a runtime type-checker and phantom-types is a category of types.

I did not know beartype supports `Annotated[basetype, <predicates>]`, but I suspect the main difference to that would be that phantom types have full compatibility with _static_ type checkers, that's really the main selling point of phantom types: to "store" results from runtime type checks in the static type system.

**Comment by antonagestam** [comment]
@majidaldo Also, I forgot to mention, phantom-types are compatible with beartype (and other runtime type checkers), as mentioned here: https://github.com/antonagestam/phantom-types#runtime-type-checking
